tests/unit/tasks/models/jobsModel.spec.ts

```diff
@@ -0,0 +1,363 @@
+      expect(findRecordMock).toHaveBeenCalledTimes(1);
+    });
+
+    it('should update job status to "Failed" if task status is "Failed"', async function () {
+      setValue('ingestionUpdateJobType', ingestionUpdateJobType);
+      setValue('ingestionTaskType', { tileMergeTask, tileSplitTask });
+
+      const rasterMapTestData = { ...testMetadata };
+      rasterMapTestData.productType = ProductType.RASTER_MAP;
+
+      getJobStatusMock.mockReturnValue({
+        id: jobId,
+        isCompleted: false,
+        isSuccessful: false,
+        percentage: 90,
+        relativePath: `test/${ProductType.RASTER_MAP}`,
+        metadata: rasterMapTestData,
+        type: ingestionUpdateJobType,
+        successTasksCount: 3,
+        status: OperationStatus.IN_PROGRESS,
+      });
+
+      getTaskMock.mockReturnValue({
+        id: taskId,
+        jobId: jobId,
+        type: tileMergeTask,
+        status: OperationStatus.FAILED,
+      });
+
+      const catalogRecordId = 'a6fbf0dc-d82c-4c8d-ad28-b8f56c685a23';
+      findRecordMock.mockResolvedValue({
+        id: catalogRecordId,
+        metadata: {},
+      });
+
+      // tasksManager = new TasksManager(
+      //   logger,
+      //   configMock,
+      //   syncClientMock,
+      //   jobManagerClientMock,
+      //   mapPublisherClientMock,
+      //   catalogClientMock,
+      //   linkBuilderMock,
+      //   metadataMergerMock
+      // );
+
+      await jobsManager.completeJob(jobId, taskId);
+
+      expect(updateJobStatusMock).toHaveBeenCalledTimes(1);
+      expect(updateJobStatusMock).toHaveBeenCalledWith(jobId, OperationStatus.FAILED, undefined, 'Failed to update ingestion');
+      expect(mergeMock).toHaveBeenCalledTimes(0);
+      expect(updateMock).toHaveBeenCalledTimes(0);
+      expect(findRecordMock).toHaveBeenCalledTimes(0);
+    });
+
+    it('should not update job status if job status is not "In-Progress"', async function () {
+      setValue('ingestionUpdateJobType', ingestionUpdateJobType);
+      setValue('ingestionTaskType', { tileMergeTask, tileSplitTask });
+
+      const rasterMapTestData = { ...testMetadata };
+      rasterMapTestData.productType = ProductType.RASTER_MAP;
+
+      getJobStatusMock.mockReturnValue({
+        id: jobId,
+        isCompleted: false,
+        isSuccessful: false,
+        percentage: 90,
+        relativePath: `test/${ProductType.RASTER_MAP}`,
+        metadata: rasterMapTestData,
+        type: ingestionUpdateJobType,
+        successTasksCount: 3,
+        status: OperationStatus.ABORTED,
+      });
+
+      getTaskMock.mockReturnValue({
+        id: taskId,
+        jobId: jobId,
+        type: tileMergeTask,
+        status: OperationStatus.COMPLETED,
+      });
+
+      const catalogRecordId = 'a6fbf0dc-d82c-4c8d-ad28-b8f56c685a23';
+      findRecordMock.mockResolvedValue({
+        id: catalogRecordId,
+        metadata: {},
+      });
+      // tasksManager = new JobsManager(
```

Comment:
why the remarks?

Reply:
Pending

should have been deleted, i removed it, old test initialized task manager every test, i made it initialize once before all and its called job manager now instad of task manager , see old tests => https://github.com/MapColonies/discrete-overseer-creator/blob/master/tests/unit/tasks/models/tasksModel.spec.ts